lib/ansible/cli/adhoc.py

```diff
@@ -20,6 +24,13 @@
 from ansible.playbook.play import Play
 from ansible.utils.display import Display
 
+# as the SSH_ASKPASS script can only be invoked as a singular command
+# python -m doesn't work here, so assume that if we are executing
+# and there is only a single item in argv plus the executable, and the
+# env var is set we are in SSH_ASKPASS mode
+if sys.argv[2:] == [] and os.getenv('_ANSIBLE_SSH_ASKPASS_SHM'):
+    _ssh_askpass.main()
+
```

Comment:
I don't really love this check.  The script is invoked with a single arg that can look like numerous things, including:

* `user@host's password: `
* `Enter PIN for 'token': `

Both of the above are things that we know we support today.  There may be others.